8315683: Parallelize java/util/concurrent/tck/JSR166TestCase.java

roy-soumadipta opened the pull request:
This test is running in tier1, and takes about 400 seconds to complete. Thus, it drags the execution time of tier1 on large machines. The commit includes changing the sequential run of test cases in java/util/concurrent/tck/JSR166TestCase.java to the best possible combination of parallelization to reduce the total runtime of the overall test and causing least possible change to user and system times. The below comparison shows existing and newer combinations of parallelizations:

* before                           : **200.64s user 20.94s system 204% cpu 1:48.47 total**
* fully parallel                  : **308.84s user 23.75s system 479% cpu 1:09.42 total**
* default-details-tck       : **244.69s user 22.03s system 314% cpu 1:24.94 total**
* default-fjp_details-others : **260.12s user 24.47s system 404% cpu 1:10.31 total**

Based on the comparison above, the fourth combination has a result very close to full parallelization and at the same time having the least deviation of system and user times among the newer combinations. Hence the commit includes the changes corresponding to the fourth combination.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315683](https://bugs.openjdk.org/browse/JDK-8315683): Parallelize java/util/concurrent/tck/JSR166TestCase.java (**Enhancement** - P4)


### Reviewers
 * [Martin Buchholz](https://openjdk.org/census#martin) (@Martin-Buchholz - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15619/head:pull/15619` \
`$ git checkout pull/15619`

Update a local copy of the PR: \
`$ git checkout pull/15619` \
`$ git pull https://git.openjdk.org/jdk.git pull/15619/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15619`

View PR using the GUI difftool: \
`$ git pr show -t 15619`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15619.diff">https://git.openjdk.org/jdk/pull/15619.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15619#issuecomment-1710211432)

Martin-Buchholz commented on test/jdk/java/util/concurrent/tck/JSR166TestCase.java:
```diff
@@ -35,31 +35,43 @@
 /*
- * @test
+ * @test id=default
  * @summary JSR-166 tck tests, in a number of variations.
```
The @summary is stale - it describes the state when there was only one @test

Martin-Buchholz commented on test/jdk/java/util/concurrent/tck/JSR166TestCase.java:
```diff
@@ -35,31 +35,43 @@
  * @build *
  * @modules java.management
  * @run junit/othervm/timeout=1000 JSR166TestCase
  * @run junit/othervm/timeout=1000 -Djava.security.manager=allow JSR166TestCase
```
security manager testing is relatively less important.   I would move this to a lower tier, while moving a test with -Djsr166.testImplementationDetails=true into tier1

Martin-Buchholz replied:
I still think the test with -Djava.security.manager=allow is relatively low value, so would also move it into its own @test id=security-manager

Martin-Buchholz reviewed (changes requested):
Hello Soumadipta - pleased to "meet" you.

If the intent is to split the tests into relatively more and less important variants for placing into separate tiers, that should be clear from the @summary's in the tests.

roy-soumadipta commented:
> Hello Soumadipta - pleased to "meet" you.
> 
> If the intent is to split the tests into relatively more and less important variants for placing into separate tiers, that should be clear from the @summary's in the tests.

Hi @Martin-Buchholz the latest commit addresses the comment regarding missing summary for each section of tests. Also from the JBS bug, I can see it has been agreed to have a separate PR for improvements to the test itself and segregation into appropriate tiers.

roy-soumadipta commented:
Hi @Martin-Buchholz I have created a separate test case for security manager as you requested. The deviations of user and system times are very less from the initial comment and with total running time improvement: **285.76s user 23.37s system 492% cpu 1:02.76 total**.
Request your re-approval on the latest changes.

Martin-Buchholz reviewed (approved):
Thanks!

roy-soumadipta commented:
/integrate

roy-soumadipta commented:
Hi @Martin-Buchholz can you please sponsor the PR

Martin-Buchholz commented:
/sponsor
